free/src/main/scala/cats/free/FreeT.scala

```diff
@@ -0,0 +1,258 @@
+package cats
+package free
+
+import scala.annotation.tailrec
+
+import cats.data.Xor
+
+/** FreeT is a monad transformer for Free monads over a Functor S
+  *
+  * This Scala implementation of `FreeT` and its usages are derived from
+  * [[https://github.com/scalaz/scalaz/blob/series/7.3.x/core/src/main/scala/scalaz/FreeT.scala Scalaz's FreeT]],
+  * originally written by Brian McKenna.
+  */
```

Comment:
It might also be good to link to [Stack Safety for Free](http://functorial.com/stack-safety-for-free/index.pdf) here. Both as attribution to Phil Freeman and because this is the paper that inspired `FreeT` arriving in Scala.